### PR TITLE
[RLlib; Offline RL] Offline RL example that shows how to customize an offline data pipeline.

### PR DESCRIPTION
**Pull request by simonsays1980**
## Why are these changes needed?

RLlib's new Offline RL was built to be more powerful as its old stack one and competitive to the SOTA standards out in the market. Part of it is its highly customizable pipeline that can be used to literally read any data from 
local or remote storage and transform data from raw format to `SingleAgentEpisode` instances to feed the learner connector pipeline and further to `MultiAgentBatch` format ready-to-train on the `Learner`. 

This example is the first of multiple ones to show users the cacabilities of RLlib's new Offline RL API. It shows 
- how to read in raw image data from a cloud bucket
- how to transform this data via byte streams to `numpy` arrays
- how to further transform these arrays into `SingleAgentEpisode` format to be readable by `ConnectorV2` pieces
- how to convert further into `MultiAgentBatch` format to pass to a `Learner`'s `update` method.

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/offline/offline_data.py [hunk 1]
```diff
@@ -223,7 +223,7 @@ def sample(
     @property
     def default_read_method_kwargs(self):
         return {
-            "override_num_blocks": max(self.config.num_learners * 2, 2),
+            # "override_num_blocks": max(self.config.num_learners * 2, 2),
```
Comment (sven1977) [inline rllib/offline/offline_data.py:226]: Is this line not needed?

File: rllib/examples/offline_rl/offline_rl_with_image_data.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+from ray.rllib.algorithms.bc.bc_catalog import BCCatalog
+from ray.rllib.algorithms.bc.torch.bc_torch_rl_module import BCTorchRLModule
+from ray.rllib.core.rl_module.rl_module import RLModuleSpec, DefaultModelConfig
+from ray.rllib.core.rl_module.multi_rl_module import MultiRLModuleSpec
+from ray.rllib.examples.offline_rl.classes.image_offline_data import ImageOfflineData
+from ray.rllib.examples.offline_rl.classes.image_offline_prelearner import (
+    ImageOfflinePreLearner,
+)
+
+# Create an Algorithm configuration.
```
Comment (sven1977) [inline rllib/examples/offline_rl/offline_rl_with_image_data.py:57]: Ok, very cool. Let's leave this for now until we have the S3 bucket publicly available in the OSS.

Let's add a TODO for now:
```
TODO: Make this an actually running/learning example with RLunplugged data from S3 and add this to the CI.
```

**Review by sven1977 — APPROVED** [review]
Looks great! Thanks for the quick turnaround @simonsays1980 !

Just the one question on the commented-out line. ??

File: rllib/offline/offline_data.py (hunk 1 above)
Comment (simonsays1980) [reply rllib/offline/offline_data.py:226]: Great catch. Somehow this makes it into multiple of my PRs.

File: rllib/examples/offline_rl/offline_rl_with_image_data.py (hunk 2 above)
Comment (simonsays1980) [reply rllib/examples/offline_rl/offline_rl_with_image_data.py:57]: We also need this other PR merged to make it run - as we do not have any way to run training without an environment.

**Pull request merged**